Slot game.part6

Fedrogan opened the pull request:
task 6 completed
1. Anticipations added
2. Added UIEffect plugin
3. Shine and pulse effects added for SCATTER symbols

Commit:
млн. replaced with млрд. in Total bonus prize popup

Commit:
task 6 completed
1. Anticipations added
2. Added UIEffects plugin
3. Shine and pulse effects added for SCATTER symbols

Commit:
Merge branch 'main' of https://github.com/Fedrogan/RidottoInternship into SLOT_GAME.Part6

# Conflicts:
#	Assets/Scenes/SampleScene.unity
#	Assets/Scripts/BonusGameController.cs
#	Assets/Scripts/Controller/AnimationsManagement.cs
#	Assets/Scripts/Events/GameController.cs
#	Assets/Scripts/Model/WinLinesCheck.cs
#	Assets/Scripts/View/ReelsScroll.cs

Commit:
merge main into Part6

Commit:
temp

Commit:
temp

Commit:
no message

Commit:
anticipations repaired after merge from main branch

MPRidotto commented on Assets/Scripts/Model/WinLinesCheck.cs:
```diff
@@ -81,4 +81,24 @@ public int CheckFSGame()
         else return 0;
     }
+
+    public bool CheckAnticipation(SubReel firstSubReel, SubReel secondSubReel)
```
Я бы вынес проверку антисипации в отдельных класс, т.к. название класса совсем о другом говорит и не подразумевает данный функционал, тоже самое с проверкой фс. Но можешь просто переименовать класс, сойдет.

Fedrogan replied:
- Проверка антисипации теперь происходит в классе ReelsScroll с помощью класса ScattersChecker(класс чекает скаттеры на всех рилах или на одном переданном). 
- WinLinesCheck теперь отвечает только за проверку линий. 
- Проверка фс происходит в классе GameModel (так по сути и было, метод CheckFSGame класса WinLinesCheck неправильно назывался, т.к. он возвращал количество скаттеров, если они были на всех трех рилах)

Commit:
Correction of deficiencies in WinLinesCheck class:
- Now the anticipation check is performed in the ReelsScroll class using the ScattersChecker class (the class checks scatters on all reels or on one).
- WinLinesCheck is now only responsible for checking lines.
- The freespins check is performed in the GameModel class (this is essentially how it was, since the CheckFSGame method of the WinLinesCheck class was is incorrectly named, since it returned the number of scatters if they were on all three reels)

MPRidotto commented:
Мерж в основную ветку
